rag_experiment_accelerator/evaluation/eval.py

```diff
@@ -269,13 +271,13 @@ def llm_answer_relevance(response_generator: ResponseGenerator, question, answer
         double: The relevancy score generated between the question and answer.
 
     """
-    try:
-        result = response_generator.generate_response(
-            sys_message=llm_answer_relevance_instruction, prompt=answer
-        )
-    except Exception as e:
-        logger.error(f"Unable to generate answer relevance score: {e}")
-        return 0
+    result = response_generator.generate_response(
```

Comment:
AFAIK `generate_response` raises exceptions.
This change removes the try/except. Why?
Do we want this function to propagate the exception upwards?

Reply:
Added note to the description.
The main idea is that now, if prompt has tag PromptTag.NonStrict in case of any exception except KeyboardInterrupt generate_response will return None without failing. Thus, except blocks are not needed anymore, you can just check response for None. If there is no NonStrict tag, then exceptions will be propagated.